docassemble/ALWeaver/data/questions/assembly_line.yml

```diff
@@ -348,9 +348,9 @@ code: |
 ---
 code: |
   if template_upload[0].filename.endswith('pdf'):
-    short_filename = template_upload[0].filename[:-4]
+    short_filename = space_to_underscore(varname(template_upload[0].filename.lower().strip(".pdf")))
```

Comment:
`.strip()` (and `rstrip`)  takes off any of given characters characters from the front and back, in any order.

I.e., `'df_form_pd.pdf'.strip('.pdf')` returns '_form_', even though 'dp_form_pd' is a fine name.

If you wanted the splice to be a bit more understandable (but slightly longer) you could try this:
```suggestion
    short_filename = space_to_underscore(varname(template_upload[0].filename[:-len(".pdf")]))
```

Reply:
Darn, that's a good catch. I hadn't given that a try yet.